Source/WebCore/rendering/svg/SVGResources.cpp

```diff
@@ -265,19 +265,19 @@ bool SVGResources::buildCachedResources(const RenderElement& renderer, const Ren
     }
 
     if (markerTags().contains(tagName) && svgStyle.hasMarkers()) {
-        AtomString markerStartId(svgStyle.markerStartResource());
+        auto markerStartId = SVGURIReference::fragmentIdentifierFromIRIString(svgStyle.markerStartResource(), document);
         if (setMarkerStart(getRenderSVGResourceById<RenderSVGResourceMarker>(document, markerStartId)))
```

Comment:
Should not this be refactored into a helper function or a template specialization instead of repeating the same call?

```
template<>
Renderer* getRenderSVGResourceById<RenderSVGResourceMarker>(Document& document, const AtomString& id)
{
    auto markerId = SVGURIReference::fragmentIdentifierFromIRIString(id, document);
    RenderSVGResource* container = getRenderSVGResourceContainerById(document, markerId);
    if (is<Renderer>(container))
        return downcast<Renderer>(container);
    return nullptr;
}
```

Reply:
I'll look into that.